### PR TITLE
Add support for arbitrary metadata in TestJob objects

### PR DESCRIPTION
**Pull request by terceiro**

### CODE REVIEW

**Comment by terceiro** [comment]
Looking at the commits individually will probably make the review easier

**Comment by terceiro** [comment]
hah, missed a setup step in the travis build

File: benchmarks/tests.py [hunk 1]
```diff
@@ -425,3 +425,45 @@ class TestJobTestCase(TestCase):
...
+        self.assertEqual(job.metadata['baz'], 'qux')
+
+    def test_extract_metadata_from_nested_metadata(self):
+        job = TestJob(definition='{ "something": { "name": "foobar", "metadata" : { "foo": "bar", "baz": "qux" } } }')
+        job.__extract_metadata__()
+
+        self.assertEqual(job.metadata['foo'], 'bar')
+        self.assertEqual(job.metadata['baz'], 'qux')
+
+    def test_extract_metadata_from_lava_job(self):
```
Comment (mwasilew) [inline benchmarks/tests.py]: should we add test for extracting metadata from more than one action in the job?

**Comment by mwasilew** [comment]
Looks good to me. The missing test might be added later.

**Comment by terceiro** [comment]
added a new test just in case

**Pull request merged**